python/paddle/distributed/auto_parallel/api.py

```diff
@@ -90,9 +91,324 @@ def sharding_specs(self):
+            )
+        if mode is None:
+            mode = self._mode
+        if mode not in ["train", "eval", "predict"]:
+            raise ValueError("mode can only be 'train', 'eval' or 'predict'.")
+        return mode
+
+    def dist_main_program(self, mode=None):
+        """
+        Get the distributed main program of ``mode``. The ``mode``
+        arg can be "train", "eval" or "predict", if it is not set,
+        ``self._mode`` will be used.
+        """
+        mode = self.__validate_mode(mode)
+        return self._engine.get_dist_main_program(mode)
+
+    def dist_startup_program(self, mode=None):
+        """
+        Get the distributed startup program of ``mode``. The ``mode``
+        arg can be "train", "eval" or "predict", if it is not set,
+        ``self._mode`` will be used.
+        """
+        mode = self.__validate_mode(mode)
+        return self._engine.get_dist_startup_program(mode)
+
+    def serial_main_program(self, mode=None):
+        """
+        Get the serial main program of ``mode``.
+        """
+        mode = self.__validate_mode(mode)
+        return self._engine.get_serial_main_program(mode)
+
+    def serial_startup_program(self, mode=None):
+        """
+        Get the serial startup program of ``mode``.
+        """
+        mode = self.__validate_mode(mode)
+        return self._engine.get_serial_startup_program(mode)
+
+    def _make_feeds(self, data_list):
+        if (
+            self._mode not in self._feed_name_list
+            or self._feed_name_list[self._mode] == []
+        ):
+            feed_list = self._engine.get_feed_list()
+            self._feed_name_list[self._mode] = [var.name for var in feed_list]
+        feed_name_list = self._feed_name_list[self._mode]
+        if len(feed_name_list) != len(data_list):
+            raise ValueError(
+                "The input data and feed_list are not consistent."
+                "The model takes %s as input" % (str(feed_name_list))
+            )
+        return dict(zip(feed_name_list, data_list))
+
+    def __call__(self, *args):
+        if self._mode is None:
+            raise ValueError("Please call train()/eval()/predict() first.")
+        if self._mode == "train":
+            if self._engine._optimizer is None or self._engine._loss is None:
+                raise ValueError(
+                    "Please set optimizer and loss function before training."
+                )
+        if self._mode == "eval":
+            if self._engine._loss is None:
+                raise ValueError("Please set loss function before evaluation.")
+        feeds = self._make_feeds(list(args))
+        outs = self._engine.run(feeds)
+        if self._mode == "predict":
+            return outs["outputs"]
+        else:
+            return outs["loss"]
+
+
 # Part2: DistTensor construction related APIs
 
 
+def to_static(
+    layer: paddle.nn.Layer,
+    loader=None,
+    loss=None,
+    optimizer=None,
+    strategy=None,
+):
```

Comment:
I saw in the design document that there is parameter of `metrics`. Shall we need to implement `metrics` which is not implemented here? If not, please explain the reason and modify the design document.

Reply:
``metrics`` is not consisted in the original design, it is not used now. So removed it here. I will modify the design document.